overlord/restart/restart.go

```diff
@@ -56,6 +57,37 @@ const (
 	RestartSystemPoweroffNow
 )
 
+type RestartBoundaryPath int
+
+const (
+	RestartBoundaryPathDo   RestartBoundaryPath = 1 << iota
+	RestartBoundaryPathUndo RestartBoundaryPath = 1 << iota
+)
+
+func restartBoundaryPaths(value string) RestartBoundaryPath {
+	var boundaries RestartBoundaryPath
+	var tokens = strings.Split(value, "|")
+	for _, t := range tokens {
+		if t == "do" {
+			boundaries |= RestartBoundaryPathDo
+		} else if t == "undo" {
+			boundaries |= RestartBoundaryPathUndo
+		}
+	}
+	return boundaries
+}
+
+func (rb *RestartBoundaryPath) String() string {
```

Comment:
shouldn't we maybe give RestartBoundaryDirection also MarshalJSON and UnmarshalJSON methods?